python/cudf/cudf/core/dataframe.py

```diff
@@ -3785,6 +3785,60 @@ def to_csv(
             chunksize,
         )
 
+    def scatter_by_map(self, map_index, map_size=None):
+        """Scatter to a list of dataframes.
+
+        Uses map_index to determine the destination
+        of each row of the original DataFrame.
+
+        Parameters
+        ----------
+        map_index : Series, str or list-like
+            Scatter assignment for each row
+        map_size : int
+            Length of output list. Must be >= uniques in map_index
+
+        Returns
+        -------
+        A list of cudf.DataFrame objects.
+        """
+
+        # map_index might be a column name or array,
+        # make it a Series
+        if isinstance(map_index, str):
+            map_index = self[map_index]
+        else:
+            map_index = Series(map_index)
+
+        # Convert float to integer
+        if map_index.dtype == np.float:
+            map_index = map_index.astype(np.int32)
+
+        # Convert string or categorical to integer
+        if isinstance(map_index._column, StringColumn):
+            map_index = Series(
+                map_index._column.as_categorical_column(np.int32).as_numerical
+            )
+        elif isinstance(map_index._column, CategoricalColumn):
+            map_index = Series(map_index._column.as_numerical)
```

Comment:
Is there ever a situation where a StringColumn or CategoricalColumn would be passed to this function?

Reply:
Since this is new functionality, there is certainly not an existing case where a StringColumn or CategoricalColumn needs to be passed here.  The StringColumn/CategoricalColumn input is handled in case someone finds it useful to partition a DataFrame into a list of DataFrames using a non-integer column.  I am happy to restrict the input to an integer types if you think it is unecessary to cover other types.

Reply:
Lets throw a warning if someone passes a String or Categorical for now that it would be more efficient to use integers and we can re-evaluate based on feedback / usage.